File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {

  static int n, m, v;
  static int[][] matrix;
  static boolean[] visited;
  static StringBuilder sb = new StringBuilder();

  public static void main(String[] args) throws IOException {

    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st = new StringTokenizer(br.readLine());
    n = Integer.parseInt(st.nextToken()); // 정점개수
    m = Integer.parseInt(st.nextToken()); // 간선개수
    v = Integer.parseInt(st.nextToken()); // 시작노드

    matrix = new int[n + 1][n + 1];
    visited = new boolean[n + 1];

    for (int i = 0; i < m; i++) {
      st = new StringTokenizer(br.readLine());
      int a = Integer.parseInt(st.nextToken());
      int b = Integer.parseInt(st.nextToken());

      matrix[a][b] = matrix[b][a] = 1;
    }

    dfs(v);
    sb.append("\n");

    visited = new boolean[n + 1];
    bfs(v);

    System.out.println(sb);

  }

  private static void dfs(int start) {

    sb.append(start).append(" ");
    visited[start] = true;

    for (int i = 1; i < n + 1; i++) {
      if (matrix[start][i] == 1 && !visited[i]) {
        dfs(i);
      }
    }

  }

  private static void bfs(int start) {
    Queue<Integer> queue = new LinkedList<>();
    queue.add(start);
    sb.append(start).append(" ");
    visited[start] = true;

    while (!queue.isEmpty()) {
      int cur = queue.poll();

      for (int i = 1; i < n + 1; i++) {
        if (matrix[cur][i] == 1 && !visited[i]) {
          visited[i] = true;
          queue.add(i);
          sb.append(i).append(" ");
        }
      }
    }
  }

}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.kt
fun main() {
    val (n, m, v) = readln().split(" ").map { it.toInt() }
    val graph = Array(n + 1) { mutableListOf<Int>() }
    var visited = BooleanArray(n + 1)
    val sb = StringBuilder()
    repeat(m) {
        val (a, b) = readln().split(" ").map { it.toInt() }

        graph[a].add(b)
        graph[b].add(a)
    }

    for (list in graph) {
        list.sort()
    }

    fun dfs(cur: Int) {
        sb.append(cur).append(' ')
        visited[cur] = true

        for (next in graph[cur]) {
            if (!visited[next]) {
                dfs(next)
            }
        }
    }

    dfs(v)
    sb.appendLine()

    visited = BooleanArray(n + 1)
    fun bfs(start: Int) {
        val queue = ArrayDeque<Int>()
        queue.add(start)

        visited[start] = true

        while (queue.isNotEmpty()) {
            val cur = queue.removeFirst()
            sb.append(cur).append(' ')

            for (next in graph[cur]) {
                if (!visited[next]) {
                    visited[next] = true
                    queue.addLast(next)
                }
            }
        }
    }

    bfs(v)

    print(sb)
}